apps/site/src/app/(main)/portal/@applicant/RsvpForm.tsx

```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import Button from "@/lib/components/Button/Button";
+
+interface RsvpFormProps {
+	buttonText: string;
+	showMessage: boolean;
+}
+
+export default function RsvpForm({ buttonText, showMessage }: RsvpFormProps) {
+	const confirmationMessage =
+		"WARNING: You will not be able to RSVP again. Are you sure you want to continue?";
+
+	return (
+		<form
+			method="post"
+			action="/api/user/rsvp"
+			onSubmit={(event) => {
+				if (showMessage && !confirm(confirmationMessage)) {
+					event.preventDefault();
+				}
+			}}
+		>
+			<Button className="text-color-red" text={buttonText} />
```

Comment:
Issue: On mobile views, the button extends past the width of the screen. Could we add a `style` prop to `Button` with a default value of `width: max-content` (so we don't affect other `Button`s) and then for this instance, pass in `width: 100%`?